### PR TITLE
Feature/mc calculation of infectiousness expectation

### PR DESCRIPTION
**Pull request by bgmoshe**

### CODE REVIEW

**Comment by bgmoshe** [comment]
Attaching here a csv that I build.
I ran the simulation through different population size and dumped the results into the csv.
It can be used to see the converges of appropriate parameters
image
image

**Comment by bgmoshe** [comment]
[MonteKlalLoConvergence.xlsx](https://github.com/CoronaHakab/CoronaHakabModel/files/4472461/MonteKlalLoConvergence.xlsx)

**Comment by bgmoshe** [comment]
> Attaching here a csv that I build.
> I ran the simulation through different population size and dumped the results into the csv.
> It can be used to see the converges of appropriate parameters
> image
> image

@meirhalachmi

File: src/corona_hakab_model/medical_state_manager.py [hunk 1]
```diff
@@ -25,16 +32,21 @@ def step(self, new_sick: List[Agent]):
...
-            agent.set_medical_state_no_inform(self.manager.medical_machine.get_state_upon_infection(agent))
+            if update_only_medical_state:
+                agent.medical_state = self.medical_state_machine.get_state_upon_infection(agent)
+            else:
+                agent.set_medical_state_no_inform(self.medical_state_machine.get_state_upon_infection(agent))
             changed_state_introduced[agent.medical_state].append(agent)
 
         # saves this number for supervising
-        self.manager.new_sick_counter = len(new_sick)  # TODO should be handled in SimulationManager
+        new_sick_counter = len(new_sick)  # TODO should be handled in SimulationManager
```
Comment (meirhalachmi) [inline src/corona_hakab_model/medical_state_manager.py:42]: remove the todo, you just did.

Comment (bgmoshe) [reply src/corona_hakab_model/medical_state_manager.py:42]: lol,  you are def right!

File: src/corona_hakab_model/manager.py [hunk 2]
```diff
@@ -76,7 +76,7 @@ def __init__(
         self.update_matrix_manager = update_matrix.UpdateMatrixManager(self)
         self.infection_manager = infection.InfectionManager(self)
         self.healthcare_manager = healthcare.HealthcareManager(self)
-        self.medical_state_manager = MedicalStateManager(self)
+        self.medical_state_manager = MedicalStateManager(self.medical_machine)
```
Comment (mor-m-shemesh) [inline src/corona_hakab_model/manager.py:79]: We want the MedicatStateManager to be aware of objects held by the SimulationManager, like the agents list and contagiousness vector.

File: src/corona_hakab_model/medical_state_manager.py [hunk 3]
```diff
@@ -11,11 +11,18 @@ class MedicalStateManager:
     Manages the medical state
     """
 
-    def __init__(self, sim_manager: "manager.SimulationManager"):
-        self.manager = sim_manager
+    def __init__(self, medical_state_machine: MedicalStateMachine):
```
Comment (mor-m-shemesh) [inline src/corona_hakab_model/medical_state_manager.py:14]: Notice here too about the manager's initialization of the MedicalStateManager

File: src/corona_hakab_model/medical_state_manager.py [hunk 4]
```diff
@@ -11,11 +11,18 @@ class MedicalStateManager:
...
+        self.medical_state_machine = medical_state_machine
         self.pending_transfers = PendingTransfers()
 
-    def step(self, new_sick: List[Agent]):
+    def step(self, new_sick: List[Agent], update_only_medical_state: bool = False):
+        """
+
+        :param new_sick: List of new agents that got sick
+        :param update_only_medical_state: Defaults to False. If True, update agent.medical_state directly.
+                                          If False it also updates agent.manager.
```
Comment (mor-m-shemesh) [inline src/corona_hakab_model/medical_state_manager.py:23]: It seems that you want to access the manager through the agents, which looks kind of twisted to me.
In the flow of 'update_only_medical_state' just ignore the manager and leave it as None

File: src/corona_hakab_model/manager.py (hunk 2 above)
Comment (bgmoshe) [reply src/corona_hakab_model/manager.py:79]: Currently he has no use for the manager. If such a connection would occur in the future it should be considered to use a different architecture and not have medial state machien have manager as a field.

File: src/corona_hakab_model/analyzers/state_machine_analysis.py [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+                * average_time_to_terminal -Empirical mean time until death/recovery
+
+    """
+    if 'consts_file' in configuration:
+        consts = Consts.from_file(configuration['consts_file'])
+    else:
+        consts = Consts()
+
+    if "circle_consts_file" in configuration:
+        consts = CirclesConsts.from_file(configuration['circle_consts_file'])
```
Comment (tal-nc) [inline src/corona_hakab_model/analyzers/state_machine_analysis.py:107]: should be circle_const, no?

Comment (bgmoshe) [reply src/corona_hakab_model/analyzers/state_machine_analysis.py:107]: you are right, I fixed this but forgot to push

**Pull request merged**